sklearn/preprocessing/tests/test_target_encoder.py

```diff
@@ -22,32 +22,64 @@
 )
 
 
-def _encode_target(X_ordinal, y_numeric, n_categories, smooth):
+def _encode_target(X_ordinal, y_numeric, n_categories, smooth, sample_weight=None):
     """Simple Python implementation of target encoding."""
     cur_encodings = np.zeros(n_categories, dtype=np.float64)
-    y_mean = np.mean(y_numeric)
+    if sample_weight is not None:
+        y_mean = np.average(y_numeric, weights=sample_weight)
+    else:
+        y_mean = np.mean(y_numeric)
 
     if smooth == "auto":
-        y_variance = np.var(y_numeric)
+        if sample_weight is not None:
+            y_variance = np.sum(sample_weight * (y_numeric - y_mean) ** 2) / np.sum(
+                sample_weight
+            )
```

Comment:
I think we might want to reuse `mean_variance_axis` instead of reimplementing weighted mean and variance computation in this function.

Reply:
Mean always needs to be computed. Variance is only computed in cases where `smooth=auto`. Should I just use `_incremental_mean_and_var` and always compute the both of them even if variance isn't going to be used?